core/services/head_tracker.go

```diff
@@ -447,8 +447,8 @@ func (ht *HeadTracker) handleNewHead(ctx context.Context, head models.Head) erro
 		ms := float64(elapsed.Milliseconds())
 		promCallbackDuration.Set(ms)
 		promCallbackDurationHist.Observe(ms)
-		if elapsed > ht.callbackExecutionThreshold() {
-			ht.logger.Warnw(fmt.Sprintf("HeadTracker finished processing head %v in %s which exceeds callback execution threshold of %s", number, elapsed.String(), ht.callbackExecutionThreshold().String()), "blockNumber", number, "time", elapsed, "id", "head_tracker")
+		if elapsed > ht.store.Config.HeadTimeBudget() {
```

Comment:
This is not the same as the total head budget time, its the per callback threshold. According to the comment its chosen to be 2 * block time. So I think we can keep the callbackExecutionThreshold and just have it as:

```
func (ht *HeadTracker) callbackExecutionThreshold() time.Duration {
    return ht.store.Config.HeadTimeBudget() / 2
}
```

with a similar comment.

Reply:
cc @samsondav if that makes sense

Reply:
Seems reasonable